xds/src/test/java/io/grpc/xds/internal/rbac/engine/cel/CelInterfaceTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds.internal;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.expr.v1alpha1.CheckedExpr;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.Descriptors.Descriptor;
+import io.grpc.xds.InterpreterException;
+import java.lang.String;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for Cel Interface. */
+@RunWith(JUnit4.class)
+public class CelInterfaceTest {
+  private RuntimeTypeProvider messageProvider;
+  private Dispatcher dispatcher;
+  private Interpreter interpreter;
+  private Activation activation;
+  private Object result;
+
+  @Test
+  public void setup() throws InterpreterException {
+    // Set up interpreter used in Cel library's eval function.
+    List<Descriptor> descriptors = new ArrayList<>();
+    messageProvider = DescriptorMessageProvider.dynamicMessages(descriptors);
+    dispatcher = DefaultDispatcher.create();
+    interpreter = new DefaultInterpreter(messageProvider, dispatcher);
+    // Set up activation used in Cel library's eval function.
+    Map<String, Object> map = new HashMap<>();
+    map.put("requestUrlPath", new Object());
+    map.put("requestHost", new Object());
+    map.put("requestMethod", new Object());
+    ImmutableMap<String, Object> apiAttributes = ImmutableMap.copyOf(map);
+    activation = Activation.copyOf(apiAttributes);
+    // Add a fake condition CheckedExpr that are being evaluated.
+    CheckedExpr conditions = CheckedExpr.newBuilder().build();
+    result = interpreter.createInterpretable(conditions).eval(activation);
+  }
+
+  @Test
+  public void testCelInterface() {
+    try {
+      setup();
```

Comment:
This is really weird. Why is one `@Test` calling another `@Test`? It looks like maybe `setup()` shouldn't have the `@Test` annotation.

Reply:
Thanks for the catch! Does `@Before` make more sense here?

Reply:
I honestly think the two methods should just be combined. That setup seems specific to a particular test, as it sets both the environment and expression. If it just set up some of the things, yeah, it could be a `@Before`. But you can also just assign messageProducer, dispatcher, and interpreter at their definitions (lines 37-39), outside of the method.

Reply:
I agree that it's better to combine those two methods. Thanks for the suggestion :)